src/gatsby/utils/models/index.js

```diff
@@ -0,0 +1,28 @@
+/** Model API builder
```

Comment:
why do we use `js` everywhere, not TS?

Reply:
Part of it is me not being familiar with TS, but here specifically TS has to be transpiled before running in `gatsby-node` since the build APIs are run in node and not webpacked. If we were to start using TS in the build process, we'd need add an extra transpilation step for the build code to support it.

Reply:
yes, makes sense. No need to transform this. I thought for a moment that I saw some logic (like templates) was moved from ts/tsx to js. Resolving this.